src/Orikivo/Services/TemplateBuilder.cs

```diff
@@ -1,512 +1,378 @@
-                var currencyTypePlacement = new Rectangle(currencyTypePoint, new Size(currencyWidth, currencyTypes.Height));
-
-
-                graphics.SetClip(levelPlacement);
-                graphics.DrawImage(levelIcon, levelPlacement);
-                levelPoint.X += 5;
-
-
-                graphics.SetClip(currencyPlacement);
-                graphics.DrawImage(currencyIcon, currencyPlacement);
-                currencyPoint.X += 9;
-
-                if (userCurrency > 999 && userCurrency < 999999999999)
-                {
-                    graphics.SetClip(currencyTypePlacement);
-                    graphics.DrawImage(currencyTypeIcon, currencyTypePlacement);
-                }
-
-                if (userCurrency > 999999999999)
-                {
-                    currencyString = "∞";
                 }
+                graphics.SetClip(fillExpPlacement);
+                graphics.FillRectangle(fillBrush, fillExpPlacement);
+            }
 
-                var expValue = userExp;
-                var expMaxValue = nextLevelExp;
-                var expBarWidth = expLevelWidth + levelIntCount - 1;
-                var expPoint = new Point(40 + 5, baseSpacer + 6 + 8 + 1);
-                if (gameNameCharacters.Length - firstActivityLength > 0)
-                {
-                    expPoint.Y += 5;
-                }
-                var expPlacement = new Rectangle(expPoint, new Size(expBarWidth, 2));
+            graphics.ResetClip();
 
-                
-                var fillBrush = new SolidBrush(Color.FromArgb(255, 255, 255));
-                var emptyBrush = new SolidBrush(Color.FromArgb(146, 146, 146));
-                graphics.SetClip(expPlacement);
-                graphics.FillRectangle(emptyBrush, expPlacement);
+            var gameNameSet = userActivity.Split();
+            var primaryGameName = string.Empty;
+            var secondaryGameName = string.Empty;
+            var primaryLength = 0;
+            var secondaryLength = 0;
 
-                ulong expSegment = (ulong)(expMaxValue / (ulong)expBarWidth);
-                var fillExpPlacement = new Rectangle(expPoint, new Size(0, 2));
+            var activityList = new[]
+            {
+                "playing ",
+                "streaming ",
+                "listening to ",
+                "watching "
+            };
 
-                if (expValue > expSegment)
+            if (gameNameCharacters.Length >= firstActivityLength)
+            {
+                foreach (var name in gameNameSet)
                 {
-                    for (ulong segment = 0; segment < (ulong)expBarWidth; segment++)
+                    if (primaryLength >= firstActivityLength)
                     {
-                        if (expValue > segment * expSegment)
+                        if (secondaryLength + name.Length <= secondActivityLength)
+                        {
+                            secondaryGameName += name + Strings.Space(1);
```

Comment:
I feel like `Strings.Space(n)` is only needed for a white space length that is longer than the specified text on its own. In this case, since only a single space is being used, it can be left as `" "` to remove an unneeded `using`.